sdk/main/src/ED25519PrivateKey.cc

```diff
@@ -96,16 +104,25 @@ std::unique_ptr<ED25519PrivateKey> ED25519PrivateKey::generatePrivateKey()
 //-----
 std::unique_ptr<ED25519PrivateKey> ED25519PrivateKey::fromString(std::string_view key)
 {
-  if (key.size() != KEY_SIZE * 2 + DER_ENCODED_PREFIX_HEX.size() && key.size() != KEY_SIZE * 2)
+  std::string formattedKey = key.data();
+  // Remove PEM prefix/suffix if is present and hex the base64 val
+  if (formattedKey.compare(
+        0, internal::asn1::PEM_EDPRK_PREFIX_STRING.size(), internal::asn1::PEM_EDPRK_PREFIX_STRING) == 0)
   {
-    throw BadKeyException("ED25519PrivateKey cannot be realized from input string: input string size should be " +
-                          std::to_string(KEY_SIZE * 2 + DER_ENCODED_PREFIX_HEX.size()) + " or " +
-                          std::to_string(KEY_SIZE * 2));
+    formattedKey = formattedKey.substr(internal::asn1::PEM_EDPRK_PREFIX_STRING.size(), formattedKey.size());
+
+    if (formattedKey.compare(formattedKey.size() - internal::asn1::PEM_EDPRK_SUFFIX_STRING.size(),
+                             formattedKey.size(),
+                             internal::asn1::PEM_EDPRK_SUFFIX_STRING) == 0)
+      formattedKey = formattedKey.substr(0, formattedKey.size() - internal::asn1::PEM_EDPRK_SUFFIX_STRING.size());
```

Comment:
Would you be able to wrap single-line `if` statements in braces?

Reply:
Done. Github ignoring outdate